src/sharedInterfaces/telemetry.ts

```diff
@@ -56,3 +56,30 @@ export enum TelemetryActions {
     LoadRecentConnections = "LoadRecentConnections",
     LoadAzureSubscriptions = "LoadAzureSubscriptions",
 }
+
+export enum ActivityStatus {
+    Succeeded = "Succeeded",
+    Failed = "Failed",
+    Canceled = "Canceled",
+}
+
+export type FinishActivity = (
+    activityStatus: Exclude<ActivityStatus, ActivityStatus.Failed>,
+    additionalProperties?: Record<string, string>,
+    additionalMeasurements?: Record<string, number>,
+) => void;
+
+export type FinishActivityFailed = (
+    error?: Error,
+    includeErrorMessage?: boolean,
+    errorCode?: string,
+    errorType?: string,
+    additionalProperties?: Record<string, string>,
+    additionalMeasurements?: Record<string, number>,
+) => void;
+
+export type FinishActivityObject = {
+    end: FinishActivity;
```

Comment:
I think `timeElapsed` would be useful here, too.  Sometimes you want to emit midpoint events in addition to just the starts and ends.